drivers/serial/uart_lpc11u6x.c

```diff
@@ -874,25 +855,14 @@ static const struct uart_driver_api uartx_api = {
 
 
 #define LPC11U6X_UARTX_INIT(idx)                                              \
+PINCTRL_DT_INST_DEFINE(idx);                                                  \
 									      \
 static const struct lpc11u6x_uartx_config uart_cfg_##idx = {	              \
 	.base = (struct lpc11u6x_uartx_regs *)                                \
 	DT_REG_ADDR(DT_NODELABEL(uart##idx)),			              \
 	.clock_drv_name =						      \
 	DT_LABEL(DT_PHANDLE(DT_NODELABEL(uart##idx), clocks)),		      \
-	.rx_pinmux_drv_name =                                                 \
-	DT_LABEL(DT_PHANDLE_BY_NAME(DT_NODELABEL(uart##idx), pinmuxs, rxd)),  \
-	.tx_pinmux_drv_name =                                                 \
-	DT_LABEL(DT_PHANDLE_BY_NAME(DT_NODELABEL(uart##idx), pinmuxs, txd)),  \
-	.clkid = DT_PHA_BY_IDX(DT_NODELABEL(uart##idx), clocks, 0, clkid),    \
-	.rx_pin = DT_PHA_BY_NAME(DT_NODELABEL(uart##idx), pinmuxs, rxd, pin), \
-	.rx_func = DT_PHA_BY_NAME(DT_NODELABEL(uart##idx),		      \
-				  pinmuxs, rxd, function),		      \
-	.rx_func = DT_PHA_BY_NAME(DT_NODELABEL(uart##idx), pinmuxs,	      \
-				  rxd, function),			      \
-	.tx_pin = DT_PHA_BY_NAME(DT_NODELABEL(uart##idx), pinmuxs, txd, pin), \
-	.tx_func = DT_PHA_BY_NAME(DT_NODELABEL(uart##idx), pinmuxs,	      \
-				  txd, function),			      \
+	.pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),                        \
```

Comment:
The instance number can't be used here either, this should work:
```suggestion
        .pincfg = PINCTRL_DT_DEV_CONFIG_GET(DT_NODELABEL(uart##idx)),         \
```